### PR TITLE
Creating a spatial analog wps process 

### PR DESCRIPTION
**Pull request by huard**
@bstdenis, @dbyrns
Here is a PR for spatial analogs. It offers six different metrics to compute analogs drawn from 
Grenier, P. et al. (2013) "An assessment of six dissimilarity
   metrics for climate analogs." Journal of Applied Meteorology and Climatology
   52.4, 733-752, DOI: 10.1175/JAMC-D-12-0170.1

I'd appreciate if someone could review it and merge if satisfactory. 

Requires OCGIS master branch and pywps develop branch.
The dissimilarity functions are well tested, but the ocgis and wps wrappers are not. 

@nilshempelmann, is this something you'd want to pull into bird-house? If so, let me know if you want me to make modifications, increase test coverage, documentation, etc.

### CODE REVIEW

File: flyingpigeon/processes/wps_spatial_analog.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+                         min_occurs=1,
+                         max_occurs=5,
+                         ),
+
+            LiteralInput('dist', "Distance",
+                         abstract="Dissimilarity metric comparing distributions.",
+                         data_type='string',
+                         min_occurs=0,
+                         max_occurs=1,
+                         default='kldiv',
```
Comment (dbyrns) [inline flyingpigeon/processes/wps_spatial_analog.py:79]: The default value is not part of the allowed values ?

**Comment by dbyrns** [comment]
environment.yml is set to use pywps=3.2.6 and ocgis-next=1.4.0.dev1. I'm not sure that's what you want.

**Comment by nilshempelmann** [comment]
That looks great!! 
@cehbrecht : We are currently prepaing a merge into master and will for sure take your prcess into account
ocgis-next=1.4.0. sould be set to ocgis=2.0 
and pywps is crrently causing troubles in the WPS :-(

File: flyingpigeon/processes/wps_spatial_analog.py (hunk 1 above)
Comment (huard) [reply flyingpigeon/processes/wps_spatial_analog.py:79]: Yes, allowed_values is metric, a reference to `dissimilarity.__all__` which is equal to 
['seuclidean', 'nearest_neighbor', 'zech_aslan',
           'kolmogorov_smirnov', 'friedman_rafsky',
           'kldiv']

**Comment by huard** [comment]
I've tried to regenerate a new environment.yml but it bears little resemblance to the current one.  Do you change it by hand ?
On my end, pywps is set to 4.0.0 in my environment.yml

**Comment by huard** [comment]
@nilshempelmann What do you mean by "pywps is crrently causing troubles in the WPS" ?

**Comment by dbyrns** [comment]
Usually I effectively change it by hand. It looks like you finally succeed too.

**Comment by dbyrns** [comment]
From my perspective the review is completed :shipit:

**Pull request merged**